docs/docs/guides/generator.mdx

```diff
@@ -56,90 +58,184 @@ nodes:
         unique: true
 ```
 
-Perform these steps in the frontend.
+Load the following schema using the [infrahubctl schema](/infrahubctl/infrahubctl-schema) command.
+
+```shell
+infrahubctl schema load widgets.yml
+```
+
+```shell
+schema 'widgets.yml' loaded successfully
+1 schema processed in 8.453 seconds.
+```
+
+### Add two new nodes
+
+Perform these steps in the [Infrahub UI](http://localhost:8000).
 
 1. Create two new widget objects
     - One with the name `widget1` and count 1
     - One with the name `widget2` and count 2
-2. Create a Standard group called "widgets"
+2. Create a [Standard group](http://localhost:8000/objects/CoreGroup) called "**widgets**
 3. Add both of the created objects to the new group
 
+:::info
+Any widget object must be added to or part of the **widgets** group to be tied to the generator.
+:::
+
 ## 1. Identify the relevant data
 
 Here we define a GraphQL query that we will use to gather information.
 
-```graphql
-query Widgets($name: String!) {
-  TestWidget(name__value: $name) {
-    edges {
-      node {
-        name {
-          value
+<Tabs groupId="convertResponse">
+  <TabItem value="Non-converted query response">
+    ```graphql title="example_generator/widget_query.gql"
+    query Widgets($name: String!) {
+      TestWidget(name__value: $name) {
+        edges {
+          node {
+            name {
+              value
+            }
+            count {
+              value
+            }
+          }
         }
-        count {
-          value
+      }
+    }
+    ```
+  </TabItem>
+  <TabItem value="Converted query response">
+    Here we must provide `__typename` and `id` within the query so the SDK can convert the query to the correct type and properly stored in the SDK.
+
+  ```graphql title="example_generator/widget_query.gql"
+  query Widgets($name: String!) {
+    TestWidget(name__value: $name) {
+      edges {
+        node {
+          __typename
+          id
+          name {
+            value
+          }
+          count {
+            value
+          }
         }
       }
     }
   }
-}
-```
+  ```
+
+  </TabItem>
+</Tabs>
 
 Create a local directory on your computer where we will store the generator files.
 
 ```shell
 mkdir example_generator
 ```
 
-Within that directory store the above GraphQL query as widget_query.gql.
-
 ## 2. Create a Python Generator
 
 The Generator class needs to implement a `generate` function that receives a `data` parameter that contains the response from the GraphQL query.
 
 The goal of this generator will be to create a number of resources that depends on the set count of the widgets.
 
```

Comment:
Can we add a bit of an explanation here on the on `convert_query_response`? What the difference is in term of the generator code.

Reply:
I agree with Wim, there is the code just under but they may need an extra info box to explain what they are gonna read.

Reply:
I moved this stuff around a bit to make it a bit easier to understand by creating the `.infrahub.yml` first and then do the explanation of the convert query response there. Let me know if that addresses your feedback.